src/protocol.rs

```diff
@@ -1,12 +1,73 @@
 use crate::broadcast::Broadcast;
 use serde::{Deserialize, Serialize};
+use std::net::SocketAddr;
 
 type SeqNum = u32;
 
-#[derive(Serialize, Deserialize, Debug, Eq, PartialEq, Clone)]
-pub enum Packet {
-    Ping(SeqNum, Vec<Broadcast>),
-    Ack(SeqNum, Vec<Broadcast>),
-    PingReq(SeqNum, Vec<Broadcast>),
-    NAck(SeqNum, Vec<Broadcast>),
+#[derive(Debug, Eq, PartialEq, Clone)]
+pub struct Packet {
+    addr: SocketAddr,
+    seq: SeqNum,
+    kind: PacketKind,
+    broadcasts: Vec<Broadcast>,
```

Comment:
Do we still want this to be a `Broadcast`? Since it's the `LimitedBroadcast` type which is actually being sent across and stored in the queue right?

Reply:
Ummmm, Im not sure yet, we can change this in the future its just a place holder for now.